pyscript.core/src/stdlib/pyweb/pydom.py

```diff
@@ -0,0 +1,420 @@
+
+        if self._element.parentElement:
+            self._parent = self.__class__(self._element.parentElement)
+
+        return self._parent
+
+    @property
+    def __class(self):
+        return self.__class__ if self.__class__ != PyDom else Element
+
+    def create(self, type_, is_child=True, classes=None, html=None, label=None):
+        js_el = js.document.createElement(type_)
+        element = self.__class(js_el)
+
+        if classes:
+            for class_ in classes:
+                element.add_class(class_)
+
+        if html is not None:
+            element.html = html
+
+        if label is not None:
+            element.label = label
+
+        if is_child:
+            self.append(element)
+
+        return element
+
+    # -------- Boilerplate Proxy for the Element API -------- #
+    @property
+    def appendChild(self):
+        return self._element.appendChild
+
+
+class Element(BaseElement):
+    def append(self, child):
+        # TODO: this is Pyodide specific for now!!!!!!
+        # if we get passed a JSProxy Element directly we just map it to the
+        # higher level Python element
+        if isinstance(child, JsProxy):
+            return self.append(self.from_js(child))
+
+        elif isinstance(child, Element):
+            self.appendChild(child._element)
+
+            return child
+
+    def from_js(self, js_element):
+        return self.__class__(js.tagName, parent=self)
+
+    def query(self, selector):
+        """The querySelector() method of the Element interface returns the first
+        element that is a descendant of the element on which it is invoked that
+        matches the specified group of selectors.
+        """
+        return self.__class__(self._element.querySelector(selector))
+
+    def query_all(self, selector):
+        """The querySelectorAll() method of the Element interface returns a static
+        (not live) NodeList representing a list of the document's elements that
+        match the specified group of selectors.
+        """
+        for element in self._element.querySelectorAll(selector):
+            yield self.__class__(element)
+
+    # -------- Boilerplate Proxy for the Element API -------- #
+    @property
+    def html(self):
+        return self._element.innerHTML
+
+    @html.setter
+    def html(self, value):
+        self._element.innerHTML = value
+
+    @property
+    def content(self):
+        return self._element.innerHTML
+
+    @content.setter
+    def content(self, value):
+        display(value, target=self.id)
+
+    @property
+    def id(self):
+        return self._element.id
+
+    @id.setter
+    def id(self, value):
+        self._element.id = value
+
+    @property
+    def checked(self):
+        return self._element.checked
+
+    @checked.setter
+    def checked(self, value):
+        self._element.checked = value
+
+    @property
+    def value(self):
+        tag = self._element.tagName
+        if tag == "INPUT":
```

Comment:
what about TEXTAREA and all others? This is strongly coupled with standard DOM behavior .. you have **tons** of other elements to consider here, including OPTION, BUTTON, and it's hard to remember on top of my head what else. Custom Elements also enter the game and `value` might be an accessor too.